### PR TITLE
Support Proxy Configuration

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Support Proxy Configuration

### Check List
- [ ] New functionality includes sanity testing.
  - [ ] All sanity tests pass. (`$ ansible-test sanity`).
  - [ ] All unit tests pass.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).

### CODE REVIEW

File: plugins/modules/oneview_appliance_proxy_configuration.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+'''
+
+EXAMPLES = '''
+- name: Creates Proxy with HTTP protocol
+  oneview_appliance_proxy_configuration:
+    config: "{{ config }}"
+    state: present
+    data:
+      server: "16.85.88.10"
+      port: 8080
```
Comment (nabhajit-ray) [inline plugins/modules/oneview_appliance_proxy_configuration.py:64]: remove the ip

File: roles/oneview_appliance_proxy_configuration/files/oneview_config.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "ip":  "10.1.19.120",
```
Comment (VenkateshRavula) [inline roles/oneview_appliance_proxy_configuration/files/oneview_config.json:2]: remove the private data like ip address and username/password

File: roles/oneview_appliance_proxy_configuration/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+# defaults file for oneview_appliance_proxy_configuration
+config: "~/.ansible/collections/ansible_collections/hpe/oneview/roles/oneview_appliance_proxy_configuration/files/oneview_config.json"
```
Comment (VenkateshRavula) [inline roles/oneview_appliance_proxy_configuration/defaults/main.yml:3]: also add ```server_ip```, ```username```, ```password``` in defaults/main.yml

File: roles/oneview_appliance_proxy_configuration/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    state: present
+    data:
+      server: "<server_ip>"
+      port: 443
+      username: "<username>"
+      password: "<password>"
+      communicationProtocol: "HTTP"
+  delegate_to: localhost
+- debug: var=appliance_proxy_configuration
+
```
Comment (VenkateshRavula) [inline roles/oneview_appliance_proxy_configuration/tasks/main.yml:14]: Please add an example to confirm if this resource achieves idempotency.

File: plugins/modules/oneview_appliance_proxy_configuration.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            state=dict(
+                required=True,
+                choices=['present', 'absent']),
+        )
+        super().__init__(additional_arg_spec=argument_spec, validate_etag_support=True)
+        self.set_resource_object(self.oneview_client.appliance_proxy_configuration)
+
+    def execute_module(self):
+        if self.state == 'present':
+            return self.resource_present(self.RESOURCE_FACT_NAME)
```
Comment (VenkateshRavula) [inline plugins/modules/oneview_appliance_proxy_configuration.py:108]: ```resource_present``` method has both ```create``` and ```update``` methods to achieve idempotency. whereas this resource supports only ```create``` and no ```update``` method. So did you tested what happens when the same payload is posted twice..? will it behave idempotently..?

File: plugins/modules/oneview_appliance_proxy_configuration.py (hunk 1 above)
Comment (yuvirani) [reply plugins/modules/oneview_appliance_proxy_configuration.py]: Done

File: plugins/modules/oneview_appliance_proxy_configuration.py (hunk 5 above)
Comment (yuvirani) [reply plugins/modules/oneview_appliance_proxy_configuration.py]: Modified the code to achieve idempotency, ignoring password field for comparison as it is null always in get response though payload has the value for it.

**Review by chebroluharika — APPROVED** [review]

File: plugins/modules/oneview_appliance_proxy_configuration_facts.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+
+class ApplianceProxyConfigurationFactsModule(OneViewModule):
+    def __init__(self):
+        super().__init__(additional_arg_spec=dict())
+        self.set_resource_object(self.oneview_client.appliance_proxy_configuration)
+
+    def execute_module(self):
+        proxy_configuration = self.resource_client.get_all()
+        proxy_configuration = proxy_configuration.data
```
Comment (VenkateshRavula) [inline plugins/modules/oneview_appliance_proxy_configuration_facts.py:67]: you can remove this line and directly mention ```proxy_configuration.data``` in ansible_facts.

```return dict(changed=False,
                    ansible_facts=dict(appliance_proxy_configuration=proxy_configuration.data))```

File: roles/oneview_appliance_proxy_configuration_facts/README.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+| config  |   No  |  | |  Path to a .json configuration file containing the OneView client configuration. The configuration file is optional. If the file path is not provided, the configuration will be loaded from environment variables.  |
+
+## Example Playbook
+ 
+```yaml
+- hosts: all
+  collections:
+    - hpe.oneview
+  roles:
+    - hpe.oneview.oneview_appliance_proxy_configuration
```
Comment (VenkateshRavula) [inline roles/oneview_appliance_proxy_configuration_facts/README.md:24]: ```- hpe.oneview.oneview_appliance_proxy_configuration_facts```

File: tests/unit/test_oneview_appliance_proxy_configuration.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+###
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import mock
+import pytest
+
+from ansible_collections.hpe.oneview.tests.unit.utils.hpe_test_utils import OneViewBaseTest
+from ansible_collections.hpe.oneview.tests.unit.utils.oneview_module_loader import ApplianceProxyConfigurationModule, OneViewModuleException
```
Comment (VenkateshRavula) [inline tests/unit/test_oneview_appliance_proxy_configuration.py:26]: you can remove ```OneViewModuleException``` from import list when you are not using it.

File: roles/oneview_appliance_proxy_configuration_facts/README.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
+## oneview_appliance_proxy_configuration
```
Comment (VenkateshRavula) [inline roles/oneview_appliance_proxy_configuration_facts/README.md:1]: Add facts in the end

**Review by VenkateshRavula — APPROVED** [review]

**Pull request merged**